feature_engine/selection/drop_correlated_features.py

```diff
@@ -86,11 +89,27 @@ def __init__(
         missing_values: str = "ignore",
     ):
 
-        if method not in ["pearson", "spearman", "kendall"]:
+        if method not in ["pearson", "spearman", "kendall"] and not callable(method):
             raise ValueError(
-                "correlation method takes only values 'pearson', 'spearman', 'kendall'"
+                "correlation method takes only values 'pearson', 'spearman', "
```

Comment:
I know that I am to blame for having so many checks in the init file.

But Sklearn actually encourages [not to have param checks in the init](https://scikit-learn.org/stable/developers/develop.html), to have them instead in the fit. So as we make modifications/emhancements to the package, I 'd like to entertain the idea of maximising compatibility with sklearn.

pandas.corr() will work or return an error depending on whether the user passes the correct argument here (pearson, spearman, kendall or callable: https://github.com/pandas-dev/pandas/blob/v1.2.5/pandas/core/frame.py#L8325-L8412

So how about removing all param checks from init, and just leaving the work to pandas.corr() as per
https://github.com/pandas-dev/pandas/blob/v1.2.5/pandas/core/frame.py#L8325-L8412

This will make the code of this transformer simpler, and a tiny bit faster, by outsourcing the checks to pandas.

The tests that you added, should still pass? 

Thoughts?

Reply:
I agree on that, as this method call `pandas.corr` under the hood. However, one interesting thing about this approach is that user should go one step back to fix the method parameter.

```python
# Now: error raises here
transformer = DropCorrelatedFeatures(
    variables=None, method=dummy_method, threshold=0.6
)
# With change: error raises here, user goes one step back to fix method parameter
Xt = transformer.fit_transform(X)
```

Reply:
Regarding tests, no major changes,  just need to remove the TypeError raising test.

Reply:
I know. I also find that why would you go one step back if you can avoid it. But this is how sklearn is designed. The initialisation of their transformers almost never fails, only when using fit(), the transformers check that the init params are allowed.

I think it is a design strategy, to keep init as clean as possible. And they also say to ensure compatibility with grid search and pipeline. I don't think those will fail with our checks. But still, I think for now, I would like to try and follow their functionality as closely as possible.

Reply:
Also not sure TypeError needs removing. pandas.corr() should not raise it?

Reply:
sure, got it. Yes pandas.corr() will raise a `ValueError`, we were checking a `TypeError` for checking the arguments for the callable method, I've just removed that test and that check.